src/core/Tree/TreeNode.tsx

```diff
@@ -0,0 +1,192 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import React from 'react';
+import { CommonProps, getWindow, useTheme } from '../utils';
+import '@itwin/itwinui-css/css/tree.css';
+import { SvgChevronRight } from '@itwin/itwinui-icons-react';
+import { IconButton } from '../Buttons/IconButton';
+import cx from 'classnames';
+import { TreeContext } from './Tree';
+
+export type TreeNodeProps = {
+  /**
+   * Unique id of the node.
+   */
+  nodeId: string;
+  /**
+   * The main text displayed on the node.
+   */
+  label: React.ReactNode;
+  /**
+   * Small note displayed below main label.
+   */
+  sublabel?: React.ReactNode;
+  /**
+   * Icon shown before label and sublabel content.
+   */
+  icon?: JSX.Element;
+  /**
+   * Is TreeNode disabled.
+   * @default false
+   */
+  isDisabled?: boolean;
+  /**
+   * Is TreeNode expanded.
+   * @default false
+   */
+  isExpanded?: boolean;
+  /**
+   * Is TreeNode selected.
+   * @default false
+   */
+  isSelected?: boolean;
+  /**
+   * Callback fired when expanding or closing a TreeNode.
+   */
+  onNodeExpanded?: (nodeId: string, expanded: boolean) => void;
+  /**
+   * Callback fired when selecting a TreeNode.
+   */
+  onNodeSelected?: (nodeId: string, selected: boolean) => void;
+  /**
+   * Checkbox to be shown before TreeNode.
+   * If undefined checkbox will not be shown.
+   * Recommended to use Checkbox component.
+   */
+  nodeCheckbox?: React.ReactNode;
+  /**
+   * Content shown after TreeNode.
+   */
+  children?: React.ReactNode;
+} & CommonProps;
+
+/**
+ * TreeNode component to display within a Tree.
```

Comment:
Would prefer to have more detailed description and example below (maybe together with Tree component or something more understandable than just repeatance of same props name). Description could consist of explanation, how Tree works, that we expect all the state to be controlled by user and some other important points. Maybe try to put that info in the way, that user would be able to use Tree with TreeNode without actually looking into Storybook. 
Just need to separate that info between TreeNode and Tree.

Reply:
Updated both `Tree` and `TreeNode` descriptions